module3/lessons/Week2/RestfulRoutes.md

```diff
@@ -0,0 +1,209 @@
+---
+layout: page
+title:  RESTful Routes
+---
+
+## Learning Objectives
+* Review CRUD functionality
+* Understand REST as it relates to CRUD
+* Learn the 7 RESTful routes
+* Understand the different between RESTful and non-RESTful routes
+
+## But first...
+
+<details>
+    <summary>
+    What is CRUD again?
+    </summary>
+
+    Thus far, we've used CRUD functionality to **C**reate, **R**ead, **U**pdate, and **D**elete from a database.
+</details>
+
+<details>
+    <summary>
+    What is an HTTP route?
+    </summary>
+
+    An HTTP route is the code responsible for receiving and responding to an HTTP request.
+</details>
+
+<details>
+    <summary>
+    What is a RESTful route?
+    </summary>
+
+    <ul>
+        <li>A RESTful route follows a common pattern to map between HTTP routes/methods and CRUD functionality.</li>
+        <li>By the way, REST stands for **RE**presentation **S**tate **T**ransfer. This may be an interview question; remember it, just in case.</li>
+    </ul>
+</details>
+
+<details>
+    <summary>
+    What is a URI?
+    </summary>
+
+    URI stands for **U**niform **R**esource **I**dentifier. It is the part of the URL after the domain. For our purpose, the URI is the same as our route path.
+    <br><br>
+    For example, for the URL <code>https://en.wikipedia.org/wiki/Uniform_Resource_Identifier</code>:
+    <ul>
+        <li> the domain is <code>en.wikipedia.org</code> </li>
+        <li> the URI is <code>/wiki/Uniform_Resource_Identifier</code></li>
+    </ul>
+</details>
+
+## But first... (sections)
+
+<section class="answer" markdown="1">
+### What is CRUD again?
+
+Thus far, we've used CRUD functionality to **C**reate, **R**ead, **U**pdate, and **D**elete from a database.
+</section>
+
+<section class="answer" markdown="1">
+### What is an HTTP route?
+
+An HTTP route is the code responsible for receiving and responding to an HTTP request.
+</section>
+
+<section class="answer" markdown="1">
+### What is a RESTful route?
+
+* A RESTful route follows a common pattern to map between HTTP routes/methods and CRUD functionality.
+* By the way, REST stands for **RE**presentation **S**tate **T**ransfer. This may be an interview question; remember it, just in case.
+</section>
+
+<section class="answer" markdown="1">
+### What is a URI?
+
+URI stands for **U**niform **R**esource **I**dentifier. It is the part of the URL after the domain. For our purpose, the URI is the same as our route path.
+
+For example, for the URL `https://en.wikipedia.org/wiki/Uniform_Resource_Identifier`:
```

Comment:
Just realized that similar to my last question, this doesn't match the URI definition on Wikipedia where the entire thing is the URI. Might be helpful to include the image from Wikipedia here, it helps me visualize what is the URI.

Reply:
https://en.wikipedia.org/wiki/Uniform_Resource_Identifier